### ISSUE TITLE
Remark Transformer is Broke (Underscores Dont Work)

### ISSUE THREAD

**Issue by bradennapier** [gatsbyjs/gatsby#5417]
## Description

Gatsby Transformer Remark does not work right.

```
If you are running multiple rai_node docker containers, **do not** share the same `bash›${HOST_FOLDER}`, each rai_node requires its own independent files.
```

In this case it will actually completely mess up the file.  I tried in every markdown parser/playground/situation and it never fails.  Prettier automatically removes all escapes because this should not need to be escaped.  

No idea why gatsby is breaking this.  Remark directly appears to have zero issue.

There is no really good docs on how to even send config values to remark and what is actually accepted so i have no freakin clue if any of the config values i tried even do anything - like "emphasis" to "*R$&*z0er 793wr" just to see if anything at all happens -- nothing

Tried turning "pedantic" to true and "commonmark" to true as well - nothing works or does anything at all

### Steps to reproduce

```
If you are running multiple rai_node docker containers, **do not** share the same `bash›${HOST_FOLDER}`, each rai_node requires its own independent files.
```

Put that into remark - render - it breaks

![image](https://user-images.githubusercontent.com/15365418/40034405-a4635d8e-57b1-11e8-81b8-ba29431de7d2.png)

This is also not just because its in a blockquote.  It will fail if there are any underscores anywhere in the file.

![image](https://user-images.githubusercontent.com/15365418/40034418-b63f486a-57b1-11e8-97e8-a8bf8ef3b06b.png)

### Expected result

It should work like all other markdown parsers work and display the content normally:

https://hackmd.io/cA9j1q64TaqiHyE8ny9ZxA

### Actual result

It looks like:

![image](https://user-images.githubusercontent.com/15365418/40034405-a4635d8e-57b1-11e8-81b8-ba29431de7d2.png)

Instead of:

![image](https://user-images.githubusercontent.com/15365418/40034478-fcf5768a-57b1-11e8-8042-ecac42079522.png)

### Environment
* Gatsby version (`npm list gatsby`):└── gatsby@1.9.260
* gatsby-cli version (`gatsby --version`): └── gatsby@1.9.260
* Node.js version: 9.11.1
* Operating System: OSX

### File contents (if changed)

Below is the current config for the plugin.  Tried adding all those settings in options to see but they appear to be ignored and there is no documentation as-to what options work and where they can be placed.   

```
 {
      resolve: 'gatsby-transformer-remark',
      options: {
        commonmark: true,
        gsm: true,
        excerpt: true,
        pedantic: true,
        plugins: [
          {
            resolve: 'gatsby-remark-images',
            options: {
              maxWidth: 690,
              wrapperStyle: 'filter: drop-shadow(0px 0px 1px rgba(0,0,0,0.4));',
            },
          },
          {
            resolve: 'gatsby-remark-responsive-iframe',
          },
          {
            resolve: 'gatsby-remark-prismjs',
            options: {
              // This is used to allow setting a language for inline code
              // (i.e. single backticks) by creating a separator.
              // This separator is a string and will do no white-space
              // stripping.
              // A suggested value for English speakers is the non-ascii
              // character '›'.
              inlineCodeMarker: '›',
            },
          },
          'gatsby-remark-copy-linked-files',
          'gatsby-remark-autolink-headers',
          'gatsby-remark-katex',
        ],
      },
    },
```

**Comment by pieh** [comment]
> There is no really good docs on how to even send config values to remark and what is actually accepted so i have no freakin clue if any of the config values i tried even do anything - like "emphasis" to "*R$&*z0er 793wr" just to see if anything at all happens -- nothing
> 
> Tried turning "pedantic" to true and "commonmark" to true as well - nothing works or does anything at all that I can tell

It seems like plugins just doesn't accept any remark settings and all settings are hardcoded - https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-remark/src/extend-node-type.js#L75-L79 . PRs are welcome to add support for that.

As for the problem - I reproduced it using our hard coded settings:
```js
var Remark = require('remark');
var html = require('remark-html');

let remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
}).use(html)

remark.process('If you are running multiple rai_node docker containers, **do not** share the same `bash›${HOST_FOLDER}`, each rai_node requires its own independent files.', function (err, file) {
  if (err) throw err;
  console.log(String(file));
})
```
Would result in
```html
<p>If you are running multiple rai<em>node docker containers, <strong>do not</strong> share the same `bash›${HOST</em>FOLDER}`,
each rai_node requires its own independent files.</p>
```

Setting `pedantic` to `false` seems to solve this issue:
```html
<p>If you are running multiple rai_node docker containers, <strong>do not</strong> share the same <code>bash›${HOST_FOLDER}</code>, each rai_node requires its own independent files.</p>
```

I'm not very familiar with markdown format, but maybe you need to escape that underscore mid-word? Or alternative approach would be to add ability to pass options to `gatsby-transformer-remark` plugin as I mentioned earlier.

**Comment by bradennapier** [comment]
Yeah definitely need to be able to pass configuration to the plugin.  `prettier` formatter will remove backslashes by default since basically all markdown parsers are setup this way.

It is a huge pain in the ass to have to manually convert every single markdown document to be a style specific for gatsby ;-).

**Comment by jonauz** [comment]
So what is the solution? Simply close ticket and leave issue standing? Year later, still same issue existing.. and then finding this bug closed for no reason!
